File: Ananya-Chandra/q1.py

#enter the input array in the form of a string with space seperated numbers
class Solution:
    
    def sb(a, n, x):
    # Your code goes here
        start=0
        end=0
        minl=n+1#initial minimum length
        sum=0#""sum
        
        while(end<n):
            #To keep adding numbers till sum>x
            while(sum<=x and end<n):
                sum+=a[end]
                end+=1
            
            #to minimize the length
            while(start<n and sum>x):
                if (end-start)<minl :
                    minl=end-start
                    
                sum-=a[start]
                start+=1
            
        return minl    
    
    print("Enter the array(in space seperated string form\n")
    a = map(int, input().split())
    a = list(a)
    n = len(a)
    x = int(input("Enter the number\n"))
    print(sb(a, n, x))       

File: Pranjal-Singh/q1.cpp
// { Driver Code Starts

#include <bits/stdc++.h>
using namespace std;


 // } Driver Code Ends

class Solution{
  public:

    int sb(int A[], int n, int x)
    {
     int c=0;
    int sum=0; int b; int f=0; int l;
for(int i=n-1;i>=0;i--)
{ b=(n-1)-i;
 for(int j=0;j<=i;j++)
 {
     for(int k=c;k<=c+b;k++)
     {
         sum=sum+A[k];
     }
     if(sum>x)
     {
         f=1;
         break;
     }
     sum=0;
     c++;
 }
 if(f==1)
 break;
 c=0; sum=0;
 }   
 if(f==1)
 {
     l=b+1;
     return l;
 }   
  }

    
};

// { Driver Code Starts.

int main() {
	// your code goes here
	int t;
	cin>>t;
	while(t--)
	{
		int n,x;
		cin>>n>>x;
		int a[n];
		for(int i=0;i<n;i++)
		cin>>a[i];
		Solution obj;
		cout<<obj.sb(a,n,x)<<endl;
	}
	return 0;
}  // } Driver Code Ends